plugins/Sfxr/Sfxr.cpp

```diff
@@ -97,33 +97,34 @@ void SfxrSynth::resetSample( bool restart )
 	fperiod=100.0/(s->m_startFreqModel.value()*s->m_startFreqModel.value()+0.001);
 	period=(int)fperiod;
 	fmaxperiod=100.0/(s->m_minFreqModel.value()*s->m_minFreqModel.value()+0.001);
-	fslide=1.0-pow((double)s->m_slideModel.value(), 3.0)*0.01;
-	fdslide=-pow((double)s->m_dSlideModel.value(), 3.0)*0.000001;
+	const auto sv = static_cast<double>(s->m_slideModel.value());
+	const auto dsv = static_cast<double>(s->m_dSlideModel.value());
+	fslide = 1.0 - sv * sv * sv * 0.01;
+	fdslide = -dsv * dsv * dsv * 0.000001;
 	square_duty=0.5f-s->m_sqrDutyModel.value()*0.5f;
 	square_slide=-s->m_sqrSweepModel.value()*0.00005f;
-	if(s->m_changeAmtModel.value()>=0.0f)
-		arp_mod=1.0-pow((double)s->m_changeAmtModel.value(), 2.0)*0.9;
-	else
-		arp_mod=1.0+pow((double)s->m_changeAmtModel.value(), 2.0)*10.0;
-	arp_time=0;
-	arp_limit=(int)(pow(1.0f-s->m_changeSpeedModel.value(), 2.0f)*20000+32);
-	if(s->m_changeSpeedModel.value()==1.0f)
-		arp_limit=0;
+	const auto cha = static_cast<double>(s->m_changeAmtModel.value());
+	arp_mod = (cha >= 0.0)
+		? 1.0 - cha * cha * 00.9
```

Comment:
```suggestion
		? 1.0 - cha * cha * 0.9
```
Small typo. Not sure if the `0` prefix is used for octal in this case, but might as well remove it

Reply:
Can you even have an octal `double` literal? I added that zero to visually line it up with the `10.0` on the next line because it looks nice. Better safe than sorry, I guess lol

Reply:
Not sure. But it's C++, you literally cannot trust it lol